collector/src/test/java/com/google/cloud/trace/zipkin/StackdriverTraceClassRule.java

```diff
@@ -0,0 +1,149 @@
+package com.google.cloud.trace.zipkin;
+
+import com.google.common.collect.Sets;
+import com.google.devtools.cloudtrace.v1.PatchTracesRequest;
+import com.google.devtools.cloudtrace.v1.Trace;
+import com.google.devtools.cloudtrace.v1.TraceServiceGrpc;
+import com.google.devtools.cloudtrace.v1.TraceSpan;
+import com.google.protobuf.Empty;
+import io.grpc.Server;
+import io.grpc.netty.NettyServerBuilder;
+import io.grpc.stub.StreamObserver;
+import org.junit.rules.MethodRule;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.FrameworkMethod;
+import org.junit.runners.model.Statement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.SocketUtils;
+
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+
+import static java.util.Collections.unmodifiableSet;
+
+/**
+ * Starts up a local Stackdriver Trace server, listening for GRPC requests on {@link #grpcURI}.
+ *
+ * Works with both {@link org.junit.ClassRule} and {@link org.junit.Rule}
+ */
+public class StackdriverTraceClassRule extends TraceServiceGrpc.TraceServiceImplBase implements TestRule, MethodRule
```

Comment:
maybe StackdriverClassRule instead of StackdriverTraceClassRule? I don't think there's anything in this impl that requires it to be a ClassRule (since the port situation is context-specific, but doesn't impact the code in this type)